.github/actions/publish/action.yml

```diff
@@ -36,7 +36,7 @@ runs:
       uses: goreleaser/goreleaser-action@v5
       with:
         version: latest
-        args: release ${{ inputs.dry-run == 'true' && '--skip=publish' || '' }}
+        args: release ${{ inputs.dry-run-gh == true && '--skip=publish' || '' }} --config .goreleaser.yaml
```

Comment:
Added a more granular variable that decides to run the action or not and specifies the config file that says how to build the binary.